### ISSUE TITLE
Re-running release script has wrong version number message

### ISSUE THREAD

**Issue by mchowning** [Automattic/pocket-casts-android#402]
### Description

I recently ran the `new_beta_release` script, but the release "failed" on CI because Google rejected the upload, so I needed to run it again. The first time I ran it it created `7.24-rc-3(9037)`. I assumed that I would probably need to revert the version bumps that were committed during the first run, but when I ran the script a second time the message it printed again said it was creating the same `7.24-rc-3(9037)` version.

![image](https://user-images.githubusercontent.com/4656348/195431462-1f1d6902-d8ad-49b1-855a-4ed4d62920f9.png)

Actually the script created a new `7.24-rc-4(9038)` version. Let's update the message here so it accurately reflects what will be done. 

I haven't looked into this, but I suspect the issue may be because the "first" `7.24-rc-3` release was never committed to the `main` branch.

### Step-by-step reproduction instructions

1. Run the `new_beta_release` script, but make it so that the release fails on CI
2. Re-run the`new_beta_release` script
3. Observe that the script advises that it is creating rc-N, but that it is actually creating rc-N+1

### Did you search for existing bug reports?

- [X] I have searched for existing bug reports.

**Comment by mchowning** [comment]
👋  @oguzkocer ! Just wanted to check and see if I'm understanding correctly how this should work. This is the first time I've used this script so I might just have something set up wrong.

Related question, I was curious to look at what things like the `android_betabuild_prechecks()`, `android_bump_version_beta()`, and `android_get_app_version()` were doing, but I'm not sure where those are defined. Would you mind pointing me in the right direction?

**Comment by oguzkocer** [comment]
> I haven't looked into this, but I suspect the issue may be because the "first" 7.24-rc-3 release had not been committed to the main branch.

@mchowning This is correct. I _think_ it is considered a feature and not a bug, or at least it used to be. If I remember correctly, the reasoning was that we are not supposed to create a new beta until the changes from the previous beta were merged to `main`. Although I might be wrong on both accounts.. 😓 

---

All the helpers are located in the https://github.com/wordpress-mobile/release-toolkit/ repo: [android_betabuild_prechecks](https://github.com/wordpress-mobile/release-toolkit/blob/trunk/lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_betabuild_prechecks.rb), [android_bump_version_beta](https://github.com/wordpress-mobile/release-toolkit/blob/trunk/lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_bump_version_beta.rb), [android_get_app_version](https://github.com/wordpress-mobile/release-toolkit/blob/trunk/lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_get_app_version.rb)

---

Let me know if you need anything else!
